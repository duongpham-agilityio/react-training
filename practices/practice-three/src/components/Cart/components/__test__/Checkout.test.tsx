import { fireEvent, render } from '@testing-library/react';

// Components
import { Checkout, TCheckoutProps } from '..';

const onCheckout = jest.fn();
const init: TCheckoutProps = {
  total: 200,
  onCheckout,
};

const setup = (props = init) => render(<Checkout {...props} />);

describe('Checkout', () => {
  it('Match to snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('onCheckout', () => {
    const { getByRole } = setup();

    fireEvent.click(getByRole('button'));

    expect(onCheckout).toMatchSnapshot();
  });
});
