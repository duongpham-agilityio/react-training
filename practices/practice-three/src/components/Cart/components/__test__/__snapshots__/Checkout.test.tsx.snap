// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Match to snapshot 1`] = `
<div>
  <div
    class="css-jtccbr"
  >
    <h2
      class="chakra-heading css-1p1l10u"
    >
      Checkout
    </h2>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1upb9tn"
    />
    <div
      class="css-1lx593h"
    >
      <p
        class="chakra-text css-0"
      >
        Total: 
        $200
      </p>
      <button
        class="chakra-button css-lup84e"
        type="button"
      >
        Checkout
      </button>
    </div>
  </div>
</div>
`;

exports[`Checkout onCheckout 1`] = `
[MockFunction] {
  "calls": [
    [
      SyntheticBaseEvent {
        "_reactName": "onClick",
        "_targetInst": null,
        "altKey": false,
        "bubbles": true,
        "button": 0,
        "buttons": 0,
        "cancelable": true,
        "clientX": 0,
        "clientY": 0,
        "ctrlKey": false,
        "currentTarget": null,
        "defaultPrevented": false,
        "detail": 0,
        "eventPhase": 3,
        "getModifierState": [Function],
        "isDefaultPrevented": [Function],
        "isPropagationStopped": [Function],
        "isTrusted": false,
        "metaKey": false,
        "movementX": 0,
        "movementY": 0,
        "nativeEvent": MouseEvent {
          "isTrusted": false,
        },
        "pageX": undefined,
        "pageY": undefined,
        "relatedTarget": null,
        "screenX": 0,
        "screenY": 0,
        "shiftKey": false,
        "target": <button
          class="chakra-button css-lup84e"
          type="button"
        >
          Checkout
        </button>,
        "timeStamp": 1697566716611,
        "type": "click",
        "view": null,
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
